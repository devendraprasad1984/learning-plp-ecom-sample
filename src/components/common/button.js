import React from "react"
import '../../css/button.css'

const Button = props => {
    const {click, val,icon, btnClass} = props
    let iconObj=<span className="icons">{icon}</span>
    const normalButton=<a className={`btn riple click ${btnClass}`} onClick={click}>{iconObj} {val}</a>
    const iconButton=<a className={`btn riple click ${btnClass}`} onClick={click}>{val}</a>
    return <span>{icon !== undefined ? normalButton : iconButton}</span>
}

export default React.memo(Button)