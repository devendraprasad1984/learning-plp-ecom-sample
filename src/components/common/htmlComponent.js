import PropTypes from "prop-types";
import React from "react";

import {config} from "../../configs/config";

const HtmlComponent = ({ text }) => {
  return (
    <div
      dangerouslySetInnerHTML={{ __html: `${config.chars.pointArrow} ${text}` }}
    />
  );
};
HtmlComponent.propTypes = {
  text: PropTypes.string.isRequired,
};

export default HtmlComponent;
