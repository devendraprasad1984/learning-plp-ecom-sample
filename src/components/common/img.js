import React from "react";
import PropTypes from "prop-types";


const Image=props=>{
    const {src,className}=props
    return <img className={className} src={src} alt={'no image'}/>
}
Image.propTypes={
    src: PropTypes.string.isRequired,
    className: PropTypes.string
}
export default Image