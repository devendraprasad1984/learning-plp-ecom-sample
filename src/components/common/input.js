import PropTypes from "prop-types";
import React from "react";
import '../../css/input.css'

const Input = (props) => {
    const {label, type, placeholder, required} = props;
    return (
        <div className="input-wrapper">
            <input
                type={type || "text"}
                required={required}
                className="form-control"
                placeholder={placeholder || ""}
                {...props}
            />
            <label htmlFor="input" className="control-label">{label}</label>
        </div>
    );
};
Input.propTypes = {
    label: PropTypes.string.isRequired,
    type: PropTypes.string,
    placeholder: PropTypes.string,
    required: PropTypes.bool,
};

export default Input;
