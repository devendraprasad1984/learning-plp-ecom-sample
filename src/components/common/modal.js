import React from "react";

const Modalify = (props) => {
  const { children, header, id, show } = props;
  return (
    <div
      className={"modal-wrapper " + (show ? "show" : "hide")}
      id={id + "-wrapper"}
    >
      <div className="modal-content">
        <div className="right">
          <button className="btn danger margin-ud" id={id + "-close"}>
            close
          </button>
        </div>
        <h2>{header}</h2>
        <div className="text">
          <p>{children}</p>
        </div>
      </div>
    </div>
  );
};
Modalify.prototype = {};
export default Modalify;
