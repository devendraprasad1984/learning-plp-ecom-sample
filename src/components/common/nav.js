import React from "react";
import {HashRouter, Route, Switch} from "react-router-dom";

import {config} from "../../configs/config";

import NoData from "./nodata";

const Nav = (props) => {
  const homeComponent= config.menu.filter(row=>row.name.toLowerCase()==='home')[0]
  const makeAppRoutes = () => {
    return config.menu.map((item, index) => {
      let path = "/" + item.name.toLowerCase();
      let routekey = "route-item-" + index;
      if (item.component === undefined)
        return (
          <Route key={routekey} path={path}>
            <NoData type="404" />
          </Route>
        );
      return (
        <Route key={routekey} path={path}>{item.component}</Route>
      );
    });
  };
  return (
    <div>
      <HashRouter>
        <div className="margin-ud">
            <Switch>
              <Route exact path={'/'}>{homeComponent.component}</Route>
              {makeAppRoutes()}
            </Switch>
        </div>
      </HashRouter>
    </div>
  );
};

export default Nav;
