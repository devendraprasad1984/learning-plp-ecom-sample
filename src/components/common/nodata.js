import PropTypes from "prop-types";
import React from "react";

import {config} from "../../configs/config";

export default function NoData({ text, type }) {
  const displayNoData = () => {
    return <h3>{text || config.messages.NO_DATA} </h3>;
  };
  const display404 = () => {
    return (
      <div className="xred notfound center">
        <h2 className="xblue size35">
          4<span className="xgray">0</span>4
        </h2>
        <br />
        <span>Page Not Found</span>
      </div>
    );
  };
  return <div>{type === "404" ? display404() : displayNoData()}</div>;
}
NoData.propTypes = {
  text: PropTypes.string,
  type: PropTypes.string,
};
