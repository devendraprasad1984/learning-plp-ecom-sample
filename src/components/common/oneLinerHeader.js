import PropTypes from "prop-types";
import React from "react";

const OneLinerHeader = (props) => {
  return <h1>{props.title}</h1>;
};

export default OneLinerHeader;

OneLinerHeader.propTypes = {
  title: PropTypes.string,
};
