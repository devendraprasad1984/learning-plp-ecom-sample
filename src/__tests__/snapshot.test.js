import React from "react";

import {appComponents, snapshotChecker} from "./common";

//checks for the unexpected changes in component tree
//one of the important checks
describe("snapshot changes of components", () => {
    appComponents.forEach((comp) => {
        let { desc, component } = comp;
        snapshotChecker(desc, component);
    });
});
