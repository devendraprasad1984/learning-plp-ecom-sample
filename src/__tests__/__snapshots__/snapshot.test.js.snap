// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot changes of components app snapshot 1`] = `
<StrictMode>
  <Provider
    store={
      Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Connect(Main) />
  </Provider>
</StrictMode>
`;

exports[`snapshot changes of components facets snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "onStateChange": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
/>
`;

exports[`snapshot changes of components product_detail snapshot 1`] = `
<div>
  <h2
    className="right"
  >
    <a
      className="click xred"
      onClick={[Function]}
    >
      BACK
    </a>
  </h2>
</div>
`;

exports[`snapshot changes of components products snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "onStateChange": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
/>
`;
